### DISCUSSION TITLE
Loading animation freezes

### DISCUSSION THREAD

**Discussion by SessionApp** [quasarframework/quasar#13987] [General - Components / Directives / etc]
I am using the [loading](https://quasar.dev/quasar-plugins/loading) quasar plugin and it freezes on mobile when doing intensive work (like adding many markers on a leaflet map). On desktop the animation only half freezes (the circle doesn't change size anymore but still rotates).

Does anyone know if this is a known issue/where it could be coming from ?

**Comment by metalsadman** [comment]
`when doing intensive work` it's blocking the main thread that's why you are experiencing a freeze, try to move off the load somewhere in a different thread like using webworkers. see this article https://logaretm.com/blog/vuex-off-main-thread/

**Reply by SessionApp** [reply]
Thank you for your answer ! However, this is not possible to do with leaflet as the data objects that I need to alter are not cloanable

**Reply by metalsadman** [reply]
you will probably have to do away with the loader then, and seek out Leaflet plugin to help you with the slowdown.
